### PR TITLE
fix(cli): sort jobs by stage completion ratio

### PR DESCRIPTION
**Pull request by QuakeWang**
# Which issue does this PR close?

Closes #.

# Rationale for this change

The TUI jobs table sorted the stage completion column with integer division:

`completed_stages / num_stages`

This could panic when `num_stages` was zero, and it also collapsed partial ratios such as `1/4`, `2/4`, and `3/4` to the same value.

# What changes are included in this PR?

This PR updates the jobs table stage-completion sorting to compare the actual completion ratio safely, treating zero-stage jobs as `0%` complete. It also adds regression tests for ratio-based sorting and zero-stage jobs.

# Are there any user-facing changes?

Yes. Sorting the TUI jobs table by stage completion now reflects the real completion ratio and no longer panics for jobs with zero stages.

### CODE REVIEW

File: ballista-cli/src/tui/domain/jobs.rs [hunk 1]
```diff
@@ -117,9 +117,16 @@ impl JobsData {
...
             }),
             SortColumn::StagesCompleted => jobs.sort_by(|a, b| {
-                let a_stages = a.completed_stages / a.num_stages;
-                let b_stages = b.completed_stages / b.num_stages;
-                let cmp = a_stages.cmp(&b_stages);
+                let stage_completion = |job: &Job| {
+                    if job.num_stages == 0 {
+                        (0_u128, 1_u128)
+                    } else {
+                        (job.completed_stages as u128, job.num_stages as u128)
```
Comment (coderfender) [inline ballista-cli/src/tui/domain/jobs.rs]: Nit : Perhaps 128bit int is an overkill here ?

**Review by coderfender — COMMENTED** [review]
left comments

File: ballista-cli/src/tui/domain/jobs.rs (hunk 1 above)
Comment (QuakeWang) [reply ballista-cli/src/tui/domain/jobs.rs]: @coderfender Good point. I switched this to compare f64 completion ratios with total_cmp while keeping the zero-stage guard. The exact u128 fraction comparison was more defensive than this TUI sorting path needs.

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @QuakeWang !
